src/Doctrine/EntityRegenerator.php

```diff
@@ -102,7 +102,17 @@ public function regenerateEntities(string $classOrNamespace)
                 }
 
                 if (!\in_array($fieldName, $mappedFields)) {
-                    continue;
+                    $traitProperties = [];
+
+                    foreach ($classMetadata->reflClass->getTraits() as $trait) {
+                        foreach ($trait->getProperties() as $property) {
+                            $traitProperties[] = $property->getName();
+                        }
+                    }
+
+                    if (\in_array($fieldName, $traitProperties) || isset($mapping['declared'])) {
+                        continue;
+                    }
```

Comment:
Can you explain this a bit? And we'll need a test case that shows what this is fixing :)